NOnion/Exceptions.fs

```diff
@@ -10,8 +10,15 @@ type NOnionException =
     new(msg: string, innerException: Exception) =
         { inherit Exception(msg, innerException) }
 
-type GuardConnectionFailedException(innerException: Exception) =
-    inherit NOnionException("Connecting to guard node failed", innerException)
+type GuardConnectionFailedException =
+    inherit NOnionException
+
+    new(innerException: Exception) =
+        { inherit NOnionException("Connecting to guard node failed",
+                                  innerException) }
+
+    new(message: string) = { inherit NOnionException(message) }
```

Comment:
Please add "Connecting to guard node failed" before the inner message

Reply:
added